AppTests/Networking/DataUploadRequestTests.swift

```diff
@@ -0,0 +1,227 @@
+      guard let nextTek = requestBody.teks[safe: index + 1] else {
+        continue
+      }
+
+      XCTAssertGreaterThanOrEqual(tek.rollingStartNumber, nextTek.rollingStartNumber)
+    }
+  }
+
+  func testSummariesAreCapped() throws {
+    let cap = 100
+    let summaries = (0 ..< 1000).map { _ in CodableExposureDetectionSummary.mock() }
+    let requestBody = DataUploadRequest.Body(
+      teks: [],
+      province: "AA",
+      exposureDetectionSummaries: summaries,
+      maximumExposureInfoCount: 0,
+      maximumExposureDetectionSummaryCount: cap
+    )
+
+    XCTAssertEqual(requestBody.exposureDetectionSummaries.count, cap)
+  }
+
+  func testExposureInfoAreCapped() throws {
+    let cap = 100
+    let summaries = (0 ..< 1000).map { _ in CodableExposureDetectionSummary.mock() }
+    let requestBody = DataUploadRequest.Body(
+      teks: [],
+      province: "AA",
+      exposureDetectionSummaries: summaries,
+      maximumExposureInfoCount: cap,
+      maximumExposureDetectionSummaryCount: Int.max
+    )
+
+    XCTAssertEqual(requestBody.exposureDetectionSummaries.flatMap { $0.exposureInfo }.count, cap)
+  }
+
+  func testLeastRecentSummaryIsDiscarded() throws {
+    let cap = 2
+    let date = Date()
+    let oldestDate = date.addingTimeInterval(-300_000)
+    let newestDate = date.addingTimeInterval(300_000)
+
+    let oldSummary = CodableExposureDetectionSummary(
+      date: oldestDate,
+      matchedKeyCount: 1,
+      daysSinceLastExposure: 2,
+      attenuationDurations: [100, 200],
+      maximumRiskScore: 3,
+      exposureInfo: []
+    )
+    let normalSummary = CodableExposureDetectionSummary(
+      date: date,
+      matchedKeyCount: 1,
+      daysSinceLastExposure: 2,
+      attenuationDurations: [100, 200],
+      maximumRiskScore: 3,
+      exposureInfo: []
+    )
+    let recentSummary = CodableExposureDetectionSummary(
+      date: newestDate,
+      matchedKeyCount: 1,
+      daysSinceLastExposure: 2,
+      attenuationDurations: [100, 200],
+      maximumRiskScore: 3,
+      exposureInfo: []
+    )
+
+    let summaries: [CodableExposureDetectionSummary] = [normalSummary, oldSummary, recentSummary]
+
+    let requestBody = DataUploadRequest.Body(
+      teks: [],
+      province: "AA",
+      exposureDetectionSummaries: summaries,
+      maximumExposureInfoCount: 0,
+      maximumExposureDetectionSummaryCount: cap
+    )
+
+    XCTAssertEqual(Set(requestBody.exposureDetectionSummaries), Set([recentSummary, normalSummary]))
+  }
+
+  func testMostRecentInfoIsDiscardedIfSameRisk() throws {
+    let cap = 2
+    let risk = 3
+    let date = Date()
+    let oldestDate = date.addingTimeInterval(-300_000)
+    let newestDate = date.addingTimeInterval(300_000)
+
+    let oldInfo = CodableExposureInfo(
+      date: oldestDate,
+      duration: 1,
+      attenuationValue: 2,
+      attenuationDurations: [100, 200],
+      transmissionRiskLevel: 3,
+      totalRiskScore: risk
+    )
+    
```

Comment:
<!--
  0 failure: 
  1 warning:  swiftlint (0.39.2...
  
  
  DangerID: danger-id-Danger;
  File: AppTests/Networking/DataUploadRequestTests.swift;
  Line: 140;
-->

- :warning: swiftlint (0.39.2): Lines should not have trailing whitespace. (trailing_whitespace)